wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java

```diff
@@ -0,0 +1,235 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.synapse.unittest;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import java.io.*;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.ArrayList;
+
+/**
+ * Goal which touches a timestamp file.
+ *
+ * @goal touch
+ * @phase process-sources
+ */
+@Mojo(name = "SynapseUnitTest")
+public class UnitTestCasesMojo extends AbstractMojo {
+
+    @Parameter(property = "testCasesFilePath")
+    private String testCasesFilePath;
+
+    @Parameter(property = "synapseServer")
+    private SynapseServer synapseServer;
+
+    /**
+     * Execution method of Mojo class.
+     */
+    public void execute() throws MojoExecutionException {
+
+        try {
+            appendTestLogs();
+            startTestingServer();
+            testCaseRunner();
+        } catch (Exception e) {
+            throw new MojoExecutionException("Exception occurred while running test cases " + e.getMessage());
+        } finally {
+            stopTestingServer();
+        }
+    }
+
+    /**
+     * Execution method of Mojo class.
+     */
+    private void testCaseRunner() throws IOException {
+
+        ArrayList<String> synapseTestCasePaths = getTestCasesFileNamesWithPaths(testCasesFilePath);
+
+        getLog().info("Detect " + synapseTestCasePaths.size() + " Synapse test case files to execute\n");
+
+        for (String synapseTestCaseFile : synapseTestCasePaths) {
+
+            String responseFromUnitTestFramework = UnitTestClient.executeTests
+                    (synapseTestCaseFile, synapseServer.getHost(), synapseServer.getUnitTestPort());
+
+            if (responseFromUnitTestFramework != null) {
+                isAssertResults(responseFromUnitTestFramework, synapseTestCaseFile);
+            } else {
+                throw new IOException("Test case parsing failed ");
+            }
+        }
+
+    }
+
+    /**
+     * Get saved SynapseTestcaseFiles from the given destination.
+     *
+     * @param testFileFolder current files location
+     * @return list of files with their file paths
+     */
+    private ArrayList<String> getTestCasesFileNamesWithPaths(String testFileFolder) {
+
+        ArrayList<String> fileNamesWithPaths = new ArrayList<>();
+        File folder = new File(testFileFolder);
+        File[] listOfFiles = folder.listFiles();
+
+        if (listOfFiles != null) {
+            for (File file : listOfFiles) {
+                String filename = file.getName();
+                if (filename.endsWith(".xml") || filename.endsWith(".XML")) {
+
+                    fileNamesWithPaths.add(testFileFolder + "/" + filename);
```

Comment:
hardcoding "/" may not work in windows environment. Shall we use File.seperator ?

Reply:
Done